worker/src/ipfs.rs

```diff
@@ -84,8 +88,9 @@ pub unsafe extern "C" fn ocall_write_ipfs(
     let state = slice::from_raw_parts(enc_state, enc_state_size as usize);
     let cid = slice::from_raw_parts_mut(cid, cid_size as usize);
 
-    let _cid = write_to_ipfs(state);
-    cid.clone_from_slice(&_cid);
+    let mut _cid = write_to_ipfs(state);
+
+    cid.clone_from_slice(_cid.to_bytes().as_slice());
```

Comment:
What are you doing with this cid for if you're using the `_cid` for the result?

Reply:
This was already like that, I just changed the code that needed to be adapted.

Reply:
Oh nevermind, i get what you mean.